### PR TITLE
MU-1686 / MU-1687 - Sonar support

### PR DESCRIPTION
**Pull request by facundopoblete**
Docs for SonarQube support (MUnit v2.3.0)

### CODE REVIEW

File: modules/ROOT/pages/coverage-maven-concept.adoc [hunk 1]
```diff
@@ -61,13 +61,15 @@ MUnit Coverage Summary
 ====================================================================================
 ----
 
-MUnit Coverage offers three types of reports:
+MUnit Coverage offers four types of reports:
```
Comment (kbostrom) [inline modules/ROOT/pages/coverage-maven-concept.adoc:64]: When I looked it up, I saw multiple presentations of "munit coverage">
Please verify and let me know, for the Term Presentation List, which is correct.

File: modules/ROOT/pages/coverage-maven-concept.adoc [hunk 2]
```diff
@@ -61,13 +61,15 @@ MUnit Coverage Summary
...
 
-* The *Console* report is printed to the console. It works with the summary report and shows an overview of the MUnit run, with a count of all resources, flows, and processors, as well as the overall application coverage level, and warnings and errors if any.
-* The *HTML* report shows the same information but formatted to fit any web browser. +
-To access the report, browse your application folder structure to the `${application.path}/target/site/munit/coverage/` path and locate the file `summary.html` at the starting point of the report and to navigate through all the data.
-* The *JSON* report shows the same information as the HTML report, in a JSON format. +
-To access the JSON report locate the `${application.path}/target/site/munit/coverage/munit-coverage.json`  fileinside your application folder structure.
+* The console report is printed to the console. It works with the summary report and shows an overview of the MUnit run, with a count of all resources, flows, and processors, as well as the overall application coverage level, and warnings and errors if any.
```
Comment (kbostrom) [inline modules/ROOT/pages/coverage-maven-concept.adoc]: Per style ref. bullet list formatting:
```suggestion
* Console report +
The console report is printed to the console. It works with the summary report and shows an overview of the MUnit run, with a count of all resources, flows, and processors, as well as the overall application coverage level, and warnings and errors if any.
```

File: modules/ROOT/pages/coverage-maven-concept.adoc [hunk 3]
```diff
@@ -61,13 +61,15 @@ MUnit Coverage Summary
...
+* The HTML report shows the same information but formatted to fit any web browser. +
```
Comment (kbostrom) [inline modules/ROOT/pages/coverage-maven-concept.adoc]:
```suggestion
* HTML report +
The HTML report shows the same information as the console report, but formatted to fit any web browser. It is available in the `${application.path}/target/site/munit/coverage/` path.+
```

File: modules/ROOT/pages/coverage-maven-concept.adoc [hunk 4]
```diff
@@ -61,13 +61,15 @@ MUnit Coverage Summary
...
+The HTML report is available in the `${application.path}/target/site/munit/coverage/` path.
```
Comment (kbostrom) [inline modules/ROOT/pages/coverage-maven-concept.adoc]:
```suggestion
```

File: modules/ROOT/pages/coverage-maven-concept.adoc [hunk 5]
```diff
@@ -61,13 +61,15 @@ MUnit Coverage Summary
...
+* The JSON report shows the same information as the HTML report, in a JSON format. +
```
Comment (kbostrom) [inline modules/ROOT/pages/coverage-maven-concept.adoc]:
```suggestion
* JSON report
The JSON report shows the same information as the HTML report, but in a JSON format. It is availabie in the `${application.path}/target/site/munit/coverage/munit-coverage.json` path.+
```

File: modules/ROOT/pages/coverage-maven-concept.adoc [hunk 6]
```diff
@@ -61,13 +61,15 @@ MUnit Coverage Summary
...
+The JSON report is availabie in `${application.path}/target/site/munit/coverage/munit-coverage.json` path.
```
Comment (kbostrom) [inline modules/ROOT/pages/coverage-maven-concept.adoc]:
```suggestion
```

File: modules/ROOT/pages/coverage-maven-concept.adoc [hunk 7]
```diff
@@ -61,13 +61,15 @@ MUnit Coverage Summary
...
+* The SONAR report shows the same information as the HTML report, in a SonarQube format. +
+The Sonar report is available in `${application.path}/target/site/munit/coverage/sonar.xml`.
```
Comment (kbostrom) [inline modules/ROOT/pages/coverage-maven-concept.adoc]:
```suggestion
```

File: modules/ROOT/pages/munit-maven-plugin.adoc [hunk 8]
```diff
@@ -392,6 +392,20 @@ The reports can be found under `${project.build.directory}/surefire-reports`.
 
 By default, it is set to `true`.
 
+== Enabling Sonar Reports
```
Comment (kbostrom) [inline modules/ROOT/pages/munit-maven-plugin.adoc:395]: We need to stop using gerunds (Enabling) for titles.
Tasks use imperative: Enable.
Concepts use various non-gerund or imperative forms, such as "Sonar Report Enablement" or, better, "SONAR Report Support"

File: modules/ROOT/pages/munit-maven-plugin.adoc [hunk 9]
```diff
@@ -392,6 +392,20 @@ The reports can be found under `${project.build.directory}/surefire-reports`.
...
+
+MUnit has built-in support for SonarQube. No additional configuration is needed, but you can disable it if you don't need it:
```
Comment (kbostrom) [inline modules/ROOT/pages/munit-maven-plugin.adoc]:
```suggestion
MUnit has built-in support for SonarQube, with a default setting of `true` for `enableSonarReports`. No additional configuration is needed.

You can find the reports under `${project.build.directory}/sonar-reports`.

You can disable SONAR reports if you don't need them:
```
Why? Is there any advantage to disabling?

File: modules/ROOT/pages/munit-maven-plugin.adoc [hunk 10]
```diff
@@ -392,6 +392,20 @@ The reports can be found under `${project.build.directory}/surefire-reports`.
...
+.Disabling sonar reports
+[source]
+----
+<enableSonarReports>false</enableSonarReports>
+----
+
+The reports can be found under `${project.build.directory}/sonar-reports`.
+
+By default, it is set to `true`.
+
```
Comment (kbostrom) [inline modules/ROOT/pages/munit-maven-plugin.adoc:412]: omit

**Review by kbostrom — APPROVED** [review]
Edited

**Review by Cristian-Venticinque — APPROVED** [review]

**Pull request merged**